public/js/src/articleController.js

```diff
@@ -0,0 +1,23 @@
+(function(exports) {
+  function ArticleController(articleList = new ArticleList()) {
+    this.articleList = articleList
+    this.articleListView = new ArticleListView(articleList)
+  }
+
+  ArticleController.prototype.insertHTML = function() {
+    let app = document.getElementById('app')
+    app.innerHTML = this.articleListView.returnHTMLstring();
+  }
+
+  ArticleController.prototype.showSingleArticle = function() {
+    let self = this;
+    window.addEventListener('hashchange', function() {
```

Comment:
It would make more sense for me to have this as part of your constructor, since this is about initialising the event handlers, and always needs to be done when you are setting up the controller.